src/components/AutocompleteSearchBar/AutocompleteSearchBar.js

```diff
@@ -0,0 +1,161 @@
+  };
+
+  _renderSuggestions(querySuggestions) {
+    // Define actions
+    const onSuggestionSelected = async (suggestion) => {
+      await this.setState({
+        currentValue: suggestion,
+      });
+      this.clearSuggestions();
+      this.executeSearch();
+    };
+
+    //Render Element
+    if (querySuggestions.length === 0) {
+      return null;
+    }
+
+    return (
+      <ul>
+        { querySuggestions.map( text => (
+          <li 
+            onClick={ () => onSuggestionSelected(text) }
+            key={ text }
+          >{ text }</li>)
+        )}
+      </ul>
+    );
+  
+  };
+
+  _renderElement(
+    placeholder,
+    debounce,
+    debounce_time,
+    queryString,
+    querySuggestions,
+    onInputChange,
+    executeSearch
+  ) {
+    // Initialize values              
+    placeholder = placeholder || 'Type something';
+    debounce = debounce || false;
+    // Define actions
+    const onBtnSearchClick = (event, input) => {
+      executeSearch();
+    };
+    const onKeyPress = (event, input) => {
+      if (event.key === 'Enter') {
+        executeSearch();
+      }
+    };
+
+    //Render Element
+    return (
+      <div className='AutoCompleteText'>
+        <Input
+          action={{
+            content: 'Search',
+            onClick: onBtnSearchClick,
+          }}
+          fluid
+          placeholder={placeholder}
+          onChange={(event, { value }) => {
+            onInputChange(value, debounce, debounce_time);
+          }}
+          value={queryString}          
+          onKeyPress={onKeyPress}
+        />
+        { this.renderSuggestions(querySuggestions) }
+      </div>
+    );
+  }
+
+  _handleAutocompleteChange = (queryString, debounce, debounce_time) => {
+    if(queryString.length >= 3) {
+      if (debounce){
+        const debouncedSuggestions = _.debounce(
+          this.updateSuggetions, 
+          debounce_time, 
+          { leading: true, }
+        );
+
+        debouncedSuggestions(this.state.currentValue);
+      }
+      else {
+        this.updateSuggetions(this.state.currentValue);
+      }
+    }
+  };
+
+  onInputChange = async (queryString, debounce, debounce_time) => {
+    await this.setState({
+      currentValue: queryString,
+    });
+    
+    this.handleAutocompleteChange(queryString, debounce, debounce_time)
+  };
+
+  executeSearch = () => {
+    this.updateQueryString(this.state.currentValue);
+  };
+
+  render() {
+    const { placeholder, debounce, debounce_time, suggestions } = this.props;
+    return this.renderElement(
+      placeholder,
+      debounce,
+      debounce_time,
+      this.state.currentValue,
+      suggestions,
+      this.onInputChange,
+      this.executeSearch
+    );
+  }
+}
+
+AutocompleteSearchBar.propTypes = {
+  queryString: PropTypes.string,
+  updateQueryString: PropTypes.func.isRequired,
+  updateSuggetions: PropTypes.func.isRequired,
+  clearSuggestions: PropTypes.func.isRequired,
+  renderElement: PropTypes.func,
```

Comment:
I think here it is missing debounce and debounceTime params (optional or required), and a default value below.

Reply:
Ok for the missing type. As for the default, i was defining it the same way than the placeholder:

``` JavaScript
    // Initialize values
    placeholder = placeholder || 'Type something';
    debounce = debounce || false;
```

In ``_renderElement``, that same way that the ``SearchBar`` component does. Should this be changed?

Reply:
@ppanero would be nice to deload the code and move defaults to defaultProps :) Start with your new component ;)

Reply:
done